### PR TITLE
gh-126390: Support for preserving order of options and nonoption arguments in gnu_getopt()

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-126390

----
📚 Documentation preview 📚: https://cpython-previews--126393.org.readthedocs.build/

### CODE REVIEW

File: Lib/test/test_getopt.py [hunk 1]
```diff
@@ -19,21 +19,34 @@ def assertError(self, *args, **kwargs):
         self.assertRaises(getopt.GetoptError, *args, **kwargs)
 
     def test_short_has_arg(self):
-        self.assertTrue(getopt.short_has_arg('a', 'a:'))
-        self.assertFalse(getopt.short_has_arg('a', 'a'))
+        self.assertIs(getopt.short_has_arg('a', 'a:'), True)
```
Comment (gpshead) [inline Lib/test/test_getopt.py]: using assertIs for True and False checks is unusual and too specifically tied to the implementation detail that those happen to be singletons.  Add comments anywhere you do this as to _why_ assertTrue or even assertEqual would be incorrect.

A nicer non-implementation detail tied way to express this that is probably what you were intending is: assertIsInstance(thing, bool) + assertTrue/False

**Comment by serhiy-storchaka** [comment]
This PR should be merged after #126375 (where that typo was already fixed.).

File: Lib/test/test_getopt.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_getopt.py]: This is a change from #126375. It should be reviewed there.

**Comment by erlend-aasland** [comment]
> This PR should be merged after #126375 (where that typo was already fixed.).

So let's mark it as a draft to make that more obvious.

**Comment by serhiy-storchaka** [comment]
Now this PR only contains the changes for #126390.

File: Lib/getopt.py [hunk 2]
```diff
@@ -24,9 +24,6 @@
 # TODO for gnu_getopt():
 #
 # - GNU getopt_long_only mechanism
-# - allow the caller to specify ordering
-# - RETURN_IN_ORDER option
-# - GNU extension with '-' as first character of option string
```
Comment (serhiy-storchaka) [inline Lib/getopt.py:29]: This all is the same feature.

**Review by gpshead — APPROVED** [review]
nice feature!

**Comment by serhiy-storchaka** [comment]
Thank you for your review @gpshead.

**Pull request merged**